### ISSUE TITLE
Toastr not firing multiple times.

### ISSUE THREAD

**Issue by tskorupka** [Foxandxss/angular-toastr#161]
Hello,

When I got something like this

```
function send(form) {
      elearningMessage.Create({
        text: vm.message,
        conversation_id: vm.conversation.id
      }).then(function (response) {
        if (response.success !== false) {
          vm.conversation.messages.push(response);
          
          vm.message = undefined;
          form.$setPristine();

          toastr.error('Meee');
          toastr.success('Message sent');
        } else {
          toastr.error(response.error);
        }
      });
    }
```

Exact lines:

```
toastr.error('Meee');
toastr.success('Message sent');
```

Above toastr will fire each message sent with two popups.

Below toastr will fire only once when I would get an error message but only once initial, not every time when my request will go with error.

```
toastr.error(response.error);
```
???

**Comment by Foxandxss** [comment]
I am sorry, I don't follow at all. Can you reword it?

**Comment by tskorupka** [comment]
So I guess there is issue like.
If I create angular controller and fire some action by ajax and then in response I will fire toastr, message should appear, but If I will fire multiple times my AJAX, toastr is not firing again same notification on screen.

**Comment by Foxandxss** [comment]
Is the code running again? And have you configured toastr with custom options?

**Comment by tskorupka** [comment]
Default options, code is running again but toastr not firing notification again.

**Comment by Foxandxss** [comment]
With default options there is nothing that would you prevent on doing that.

I would need a reproduction but I think that the issue is elsewhere.

**Comment by n2sandhu** [comment]
+1. Having the same issue. Noticed this after having used toastr on several page in my project

**Comment by n2sandhu** [comment]
@tskorupka : found the solution to the issue.
It says in the documention that by default toastr is configured to allow duplicates of previously show toasts, but apparently something is going wrong there.
So all that needs to be done is put this bit of code in your application and it would work.

`app.config(function(toastrConfig) {
  angular.extend(toastrConfig, {
    autoDismiss: false,
    containerId: 'toast-container',
    maxOpened: 0,    
    newestOnTop: true,
    positionClass: 'toast-top-right',
    preventDuplicates: false,
    preventOpenDuplicates: false,
    target: 'body'
  });
});`

**Comment by Foxandxss** [comment]
All of those are the default, are you sure you are not overriding them elsewhere?

**Comment by tskorupka** [comment]
Also this is my default configuration.

```
toastrConfig.allowHtml = true;
    toastrConfig.timeOut = 3000;
    toastrConfig.positionClass = 'toast-top-right';
    toastrConfig.preventDuplicates = true;
    toastrConfig.progressBar = true;
```

**Comment by tskorupka** [comment]
I think I've made mistake in my configuration.

**Comment by Foxandxss** [comment]
That preventDuplicates would prevent to open the same toast twice (you need a different toast in between so you can open the same again).

I really need a reproduction with configuration to see what's going on.

**Comment by tskorupka** [comment]
@n2sandhu thank you, I found my stupidity in this issue, also if we have a solution.

Toastr configuration is
```
toastrConfig.preventDuplicates = true;
```

Binding to button those:

```
          toastr.error('Meee');
          toastr.success('Message sent');
```

Should they fire if we click on button like 3 times?
I meant if we just can make duplicates of them and they would be shown.

**Comment by Foxandxss** [comment]
Duplicate is only for one toast, so even with that option you could get:

Meee
Message sent
Meee
Message sent
Meee
Message sent

**Comment by tskorupka** [comment]
That explains everything, thank you.

**Comment by LockonF** [comment]
I had the same issue while trying to fire the same toast multiple times. My config was the following:

angular.module('app').config(function(toastrConfig) {
  angular.extend(toastrConfig, {
    autoDismiss: false,
    containerId: 'toast-container',
    maxOpened: 0,    
    newestOnTop: true,
    positionClass: 'toast-top-right',
    preventDuplicates: false,
    preventOpenDuplicates: false,
    target: 'body'
  });
});

When I inspect the loaded script (angular-tostr.tpls.js), the default config for preventDuplicates and preventOpenDuplicates is the same as the one shown on above, but having a closer look it seems that the preventDuplicate var loaded in the default config is set as true for some reason.

<img width="1440" alt="screen shot 2016-05-13 at 8 43 05 pm" src="https://cloud.githubusercontent.com/assets/7435348/15265507/52be0ade-194b-11e6-94fe-66ad3efa3fa0.png">

**Comment by NazgulTuga** [comment]
I encoutered the same problema, but the "preventDuplicates" was set to "false".
I found out that the funciont "shouldExit" was checking if the field "message" was the same as the previous.

What i could do was: remove the line:
`if (shouldExit(options, map)) { return; }`

Or generate a random number and paste it in the field "message":
`toastr.success('<i style="display:none">'+rand+'</i>', .... );`
